### ISSUE TITLE
SciPyDistribution __init__ failure with scipy 1.9.0

### ISSUE THREAD

**Issue by josephmure** [openturns/openturns#2125]
The test script [t_Distribution_scipy.py](https://github.com/openturns/openturns/blob/master/python/test/t_Distribution_scipy.py) currently fails Continuous Integration for 2 reasons:

1. The following line is now invalid:
https://github.com/openturns/openturns/blob/82c89cd7e5e92aade1b6d216167453508aaa4684/python/test/t_Distribution_scipy.py#L8
It should be replaced with a call to numpy.random.seed(42)

2. SciPyDistributions fail to initialize because of this check in the __init__:
https://github.com/openturns/openturns/blob/b5797d7e4a71c71faf86df51f26ad0d8d551ad08/python/src/Distribution.i#L143
Unfortunately, `rv_frozen` is now the parent class, but most actual scipy distributions now have `dist.__class__.__name__` yield `rv_continuous_frozen`.

The [scipy documentation](https://docs.scipy.org/doc/scipy/reference/stats.html#probability-distributions) states there are 3 distribution types in the library:
- `rv_continuous`
- `rv_discrete`
- `rv_histogram`

I think both `rv_continuous` and `rv_histogram` (which actually is a child class of `rv_continuous`) are fully compatible with our `SciPyDistribution`framework. However, `rv_discrete` does not implement the `pdf` method and implements a `pmf` method instead. Since OT Distributions always use `computePDF` regardless of whether they are continuous or discrete, maybe the `SciPyDistribution` should be updated to accomodate the diversity of scipy APIs.

**Comment by sofianehaddad** [comment]
Hi, thanks for reporting this @josephmure 
Having a look at discrete distributions, I found things strange in scipy. For example : 
```python
import scipy.stats as st
dist = st.bernoulli(0.2)
# print probability density function at points [0] &[1]
print(dist.pmf(0.0), dist.pmf(1.0))
# print probability density function at points [0.5]
print(dist.pmf(0.5))
```
Last result is quite strange. Do we need to interface discrete distributions in OT?

**Comment by josephmure** [comment]
Indeed! Apparently scipy relies for its Bernoulli `pmf` on the [Boost binomial `pdf`](https://github.com/scipy/scipy/blob/f57054ffe2b814599fdcd224d3a9768550f548c5/scipy/stats/_discrete_distns.py#L77)

 Here is a warning that appears in the [Boost Binomial documentation](https://www.boost.org/doc/libs/1_52_0/libs/math/doc/sf_and_dist/html/math_toolkit/dist/dist_ref/dists/binomial_dist.html):



**Caution**
*The Binomial distribution is a discrete distribution: internally, functions               like the cdf and pdf are treated "as if"               they are continuous functions, but in reality the results returned               from these functions only have meaning if an integer value is provided               for the random variate argument.*

This is IMO a solid argument against us supporting scipy `rv_discrete_frozen` classes in OT.

**Comment by sofianehaddad** [comment]
Probably. Let us wait @regislebrun @jschueller and others feedbacks

**Comment by jschueller** [comment]
I think we could add support for discrete distributions as long as we keep it consistent with the OT behavior: return 0 outside the support
I can be added in another PR though
